src/NUnitEngine/nunit.engine.api/IRuntimeFrameworkService.cs

```diff
@@ -45,6 +45,11 @@ public interface IRuntimeFrameworkService
         /// the settings in the package and the assemblies themselves.
         /// The package RuntimeFramework setting may be updated as a 
         /// result and the selected runtime is returned.
+        ///
+        /// Note that if a package has subpackages, the subpackages may run on a different
+        /// framework to the top-level package. In future, this method should
+        /// probably not return a simple string, and instead require runners
+        /// to inspect the test package structure, to find all desired frameworks.
         /// </summary>
         /// <param name="package">A TestPackage</param>
         /// <returns>The selected RuntimeFramework</returns>
```

Comment:
As a future breaking change to the Engine API - I'd like this method to return void. This method is quite side-effect based, and with this change, returning a simple string here is a little mis-leading.

Reply:
Unless it were only called by runners that actually __need__ a single string, e.g. the ProcessRunner. I have a vague memory that that was the case at one time.

Reply:
Ah, maybe it was! This string is currently unused within the engine.